dev-java/netty-tcnative/netty-tcnative-1.1.33.ebuild

```diff
@@ -0,0 +1,77 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI="5"
+
+WANT_AUTOCONF="latest"
+WANT_AUTOMAKE="latest"
+
+inherit autotools autotools-utils eutils java-pkg-2 java-pkg-simple
+
+DESCRIPTION="Tomcat Native fork for Netty"
+
+SLOT="0"
+SRC_URI="https://github.com/netty/${PN}/archive/${PV}.zip"
+HOMEPAGE="http://netty.io/wiki/forked-tomcat-native.html"
+KEYWORDS="~amd64 ~x86"
+LICENSE="Apache-2.0"
+IUSE=""
+
+RDEPEND="dev-libs/apr:1
+	dev-libs/openssl:=
+	>=virtual/jre-1.7"
+
+DEPEND=">=virtual/jdk-1.7
+	${RDEPEND}"
+
+S=${WORKDIR}/${P}/src/main
+
+MY_S="${S}"
+
+JAVA_SRC_DIR="java"
+
+java_prepare() {
+	rm c/os_win32* || die
+
+	cd native-package || die
+
+	mv ../c src || die
+
+	SOURCES=$(ls -w 1000 src | tr "\n" " ") || die
```

Comment:
I have not seeing anything in the devmanual that says do not use ls in an ebuild. Not to mention if you grep the tree, there are quite many doing such; postgresql, klibc, and others. If such is not documented in the devmanual then it needs to be, and others will stop doing this. -w is another issue, and really this was just a "hack" to avoid having to manually create a list of c files. Granted there are other more elegant ways to go about such, not to mention portable. Best way to set such standards is in documentation. Which if it exist, excuse me for having missed that, but to my knowledge such is not covered in the devmanual.